### PR TITLE
fv crate

### PR DESCRIPTION
**Pull request by g-r-a-n-t**
- added a `runtimeBytecode` output option. e.g. `fe foo.fe --emit runtimeBytecode`
- created a `fe-fv` crate and added some contract verification tests

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] OPTIONAL: Update [Spec](https://github.com/ethereum/fe/blob/master/docs/src/spec/index.md) if applicable
- [x] Add entry to the [release notes](https://github.com/ethereum/fe/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [x] Clean up commit history

### CODE REVIEW

File: crates/test-files/fixtures/kspecs/sanity/foo.fe [hunk 1]
```diff
@@ -0,0 +1,28 @@
+use std::evm
+
+contract Return42:
+    pub fn __call__():
+        unsafe:
+            evm::mstore(0, 42)
+            evm::return_mem(0, 32)
+
+
+contract ReturnIn:
```
Comment (g-r-a-n-t) [inline crates/test-files/fixtures/kspecs/sanity/foo.fe:10]: add  comment

File: crates/test-files/fixtures/kspecs/sanity/returns_42.k [hunk 2]
```diff
@@ -0,0 +1,104 @@
+module TEST_NAME
```
Comment (g-r-a-n-t) [inline crates/test-files/fixtures/kspecs/sanity/returns_42.k:1]: there should prly be a `$` in these identifiers

File: crates/fv/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    let kevm_path = kevm_path();
+    let spec = build_spec(name, src_path, src, spec).unwrap();
+    let spec_path = Path::new(&kevm_path)
+        .join(SPECS_DIR)
+        .join(name)
+        .with_extension("k");
+
+    fs::write(spec_path.clone(), spec).expect("Unable to write file");
+
+    let path = env::vars()
```
Comment (g-r-a-n-t) [inline crates/fv/src/lib.rs:24]: just use `var`

File: crates/fv/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+readme = "README.md"
+repository = "https://github.com/ethereum/fe"
+version = "0.13.0-alpha"
+
+[features]
+solc-backend = ["fe-driver/solc-backend"]
+kevm-backend = []
+
+[dependencies]
+indexmap = "1.6.2"
```
Comment (g-r-a-n-t) [inline crates/fv/Cargo.toml:18]: this used?

File: crates/fv/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+authors = ["The Fe Developers <snakecharmers@ethereum.org>"]
+categories = ["cryptography::cryptocurrencies", "command-line-utilities", "development-tools"]
+description = "Fe formal verification utilities"
+edition = "2021"
+keywords = ["ethereum", "fe", "yul", "smart", "contract", "compiler"]
+license = "GPL-3.0-or-later"
+name = "fv"
+readme = "README.md"
+repository = "https://github.com/ethereum/fe"
+version = "0.13.0-alpha"
```
Comment (g-r-a-n-t) [inline crates/fv/Cargo.toml:11]: version should be 14 after rebasing

File: crates/driver/src/lib.rs [hunk 6]
```diff
@@ -30,6 +30,7 @@ pub struct CompiledContract {
     pub yul: String,
     #[cfg(feature = "solc-backend")]
     pub bytecode: String,
+    pub runtime_bytecode: String,
```
Comment (g-r-a-n-t) [inline crates/driver/src/lib.rs:33]: this should be gated by a separate emit parameter

File: crates/yulgen/src/mappers/module.rs [hunk 7]
```diff
@@ -21,3 +21,20 @@ pub fn module(db: &dyn YulgenDb, module: ModuleId) -> YulContracts {
             contracts
         })
 }
+
+pub fn module_runtimes(db: &dyn YulgenDb, module: ModuleId) -> YulContracts {
```
Comment (g-r-a-n-t) [inline crates/yulgen/src/mappers/module.rs:25]: look this stuff over

**Comment by sbillig** [comment]
@g-r-a-n-t I haven't been able to get this to work yet. Followed the directions to build kevm and run the fe fv tests. First I got 
> thread 'sanity_returns_in_cond1' panicked at 'unable to write file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', crates/fv/src/lib.rs:22:40

Solved that with a `mkdir ~/src/evm-semantics/tests/specs/fe`

Then I got 
> thread 'sanity_returns_in_cond1' panicked at '[Error] Critical: Does not exist or not a directory:
/home/sean/src/evm-semantics/./tests/specs/erc20/verification/haskell

Did a `mkdir ~/src/evm-semantics/tests/specs/erc20/verification/haskell`

Then I got 
> thread 'sanity_returns_in_cond1' panicked at '[Error] Critical: Could not find a compiled definition. Use --directory to
specify one.

So I guess there should be something in `evm-semantics/tests/specs/erc20/verification/haskell` that I'm missing?

File: crates/fv/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+    let path = env::var("PATH").expect("PATH is not set");
+
+    let out = Command::new("kevm")
+        .arg("prove")
+        .arg(spec_path.to_str().unwrap())
+        .arg("--backend")
+        .arg("haskell")
+        .arg("--format-failures")
+        // we should define out own verification module
```
Comment (g-r-a-n-t) [inline crates/fv/src/lib.rs:31]: delete this

**Comment by g-r-a-n-t** [comment]
https://github.com/ethereum/fe/pull/887

**Pull request closed without merging**